File: codeforces/problemset/A. Jzzhu and Children.cpp
#include <bits/stdc++.h>
using namespace std;
struct dron{
    int num;
    int have;
    int need;
};
int main(){
    int n,m;
    cin>>n>>m;
    deque<dron>d;
    for(int i=0,x;i<n;i++)
    {
        cin>>x;
        if(x>m){
            d.push_back({i+1,m,x});
        }
    }
    
    while(d.size()>1){
         if(d.front().have+m>=d.front().need){
              d.pop_front();
         }else{
              d.push_back({d.front().num,d.front().have+m,d.front().need});
              d.pop_front();
         }
    }
    
    if(d.empty())
     cout<<n<<endl;
    else
     cout<<d.back().num;
    

    return 0;
}

File: codeforces/problemset/A. Jzzhu and Children.java
import java.util.*;

class Drone {
    int num;
    int have;
    int need;

    public Drone(int num, int have, int need) {
        this.num = num;
        this.have = have;
        this.need = need;
    }
}

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int n = scanner.nextInt();
        int m = scanner.nextInt();
        
        Deque<Drone> deque = new ArrayDeque<>();
        
        // Reading the input and initializing the deque
        for (int i = 0; i < n; i++) {
            int x = scanner.nextInt();
            if (x > m) {
                deque.addLast(new Drone(i + 1, m, x));
            }
        }
        
        // Processing the drones in the deque
        while (deque.size() > 1) {
            Drone front = deque.peekFirst();
            if (front.have + m >= front.need) {
                deque.pollFirst();
            } else {
                deque.addLast(new Drone(front.num, front.have + m, front.need));
                deque.pollFirst();
            }
        }
        
        // Output the result
        if (deque.isEmpty()) {
            System.out.println(n);
        } else {
            System.out.println(deque.peekLast().num);
        }

        scanner.close();
    }
}
